python/ray/data/dataset.py

```diff
@@ -1719,6 +1719,33 @@ def groupby(self, key: Optional[str]) -> "GroupedData":
 
         return GroupedData(self, key)
 
+    def distinct(self) -> "Dataset":
+        """Remove duplicate rows from the :class:`~ray.data.Dataset`.
+
+        Examples:
+            >>> import ray
+            >>> ds = ray.data.from_items([1, 2, 3, 2, 3])
+            >>> ds.distinct().take_all()
+            [{'item': 1}, {'item': 2}, {'item': 3}]
+
+        Time complexity: O(dataset size * log(dataset size / parallelism))
+
+        .. note:: Currently distinct only supports :class:`~ray.data.Dataset`s with one single column.
+
+        Returns:
+            A new :class:`~ray.data.Dataset` with distinct rows.
+        """
+        columns = self.columns(fetch_if_missing=True)
+        assert columns is not None
+        if len(columns) > 1:
+            # TODO(hchen): Remove this limitation once groupby supports
+            # multiple columns.
+            raise NotImplementedError(
+                "`distinct` currently only suports Datasets with one single column, "
+                "please apply `select_columns` before `distinct`."
+            )
+        return self.groupby(columns[0]).count().drop_columns(["count()"])
```

Comment:
shall we add a TODO to implement an aggregate function for distinct, so we don't need to calculate count?

Reply:
instead of `drop_columns(["count()"])`, can we call `select_columns(columns[0])`, so we don't rely on the implicit naming of `count()`?

Reply:
> shall we add a TODO to implement an aggregate function for distinct, so we don't need to calculate count?

I considered this initially. but considering count is already very cheap, this is probably no big benefit to implement a standalone distinct function.